### ISSUE TITLE
WSLg will support Windows 10 21H2???

### ISSUE THREAD

**Issue by heroku-miraheze** [microsoft/wslg#446]
My computer will not upgrade to Windows 11, and i need to know that WSLg will support Windows 10 21H2.

Thank you.

**Comment by crramirez** [comment]
Hello @heroku-miraheze ,

Sadly, no: https://github.com/microsoft/wslg/issues/347#issuecomment-877695975

Regards,
Carlos

**Comment by Ueland** [comment]
In such case, the documentation on the README.md should be updated to state that fact. It uses both Windows 10 screenshots and specifies that next major Windows version will get it, causing people to expect it to become available in Windows 10.

**Comment by exitdown** [comment]
enhancement request here https://github.com/microsoft/wslg/issues/414

**Comment by vasilvestre** [comment]
> Hello @heroku-miraheze ,
> 
> Sadly, no: [#347 (comment)](https://github.com/microsoft/wslg/issues/347#issuecomment-877695975)
> 
> Regards,
> Carlos

It feels like a joke. I've invested so much, waited for so long and I'm frustrated because X11 isn't a good solution. I can't use insider preview as it would be against my company policy (stability issues) on windows 10 and they won't ship it ?
I really hate ms rn.

**Comment by UrsaWright** [comment]
> 
> 
> Hello @heroku-miraheze ,
> 
> Sadly, no: [#347 (comment)](https://github.com/microsoft/wslg/issues/347#issuecomment-877695975)
> 
> Regards,
> Carlos

Insane that I thought this was coming to a Windows 10 build only to have that de-confirmed in a issues thread. Microsoft continues to disappoint, this time without a public statement. Shame that an anticipated feature supported and build in preview will never come to the platform it was developed for. Don't act surprised when people meet Windows 11 with the same disdain as Windows 8 at this rate.

**Comment by Masamune3210** [comment]
Do you realize how many things are built on the system proceeding the one it releases on? It's common practice. There is more work than people seem to think involved with getting a system wide X server implementation working in Windows without gigantic bugs

**Comment by UrsaWright** [comment]
Than why bother announcing it as coming to "the next build"? It sounds like it's a complete feature that was just waiting for a release not a feature that's too early to confirm or deny. I get what you mean and I know it might not be up to the devs but why would they announce and promote it in a way that is directly counter to that?

**Comment by Masamune3210** [comment]
They might have thought that they could get it done before the release feature freeze? That's what happened to Android support and winget so it wouldn't surprise me

**Comment by angelog0** [comment]
My machine, based on Ryzen 5 and 8 GiB RAM, is only 3 years old and for what I read I do not have enough hardware to upgrade to Win 11. In any case, I never will upgrade to an OS which, for what I read, will request an MS account to complete its installation. I consider this limiting the user freedom. After all, this is a dual boot machine and it is not so hard to convert the Windows partition to a Linux partition so the the it becomes a full Linux machine...

**Comment by angelog0** [comment]
Why MS is following Apple?

macOS X --> Win !0,  Apple ID...
macOS XI -- Win 11, MS account...

When Win10 was introduced it should have been the last...

**Comment by aregenscheid** [comment]
> There is something [MicrosoftDocs/WSL@a494487](https://github.com/MicrosoftDocs/WSL/commit/a49448706f7c4569c72deea4d0874755339f8bc0)

Windows 11 only. See: https://docs.microsoft.com/en-us/windows/wsl/tutorials/gui-apps

**Comment by ivucica** [comment]
Both my lack of TPM hardware (that I don't plan on addressing) and the requirement for SecureBoot (that would place various restrictions on my dual-boot Debian) mean I am not touching Windows 11 for the foreseeable future. To me as a home power user, neither TPM nor SecureBoot provide enough benefits.

Tying WSLg to Windows 11 is very unfortunate and mean I'll just keep using a standalone X11 server; I mostly ignore Wayland anyway.

**Comment by trallnag** [comment]
Maybe we can push Microsoft to drop support for W10 as soon as possible 😄 This way employers may be forced to upgrade faster to W11

**Comment by Biswa96** [comment]
Testing WSLg (kind of) in my Windows 10 21H1 x64 system. Need to figure out many stuff.

![Capture](https://user-images.githubusercontent.com/31443074/142042841-56772bd1-e058-4120-89c1-c85e3b16e6e9.PNG)

**Comment by angelog0** [comment]
@Biswa96,

> Testing WSLg (kind of) in my Windows 10 21H1 x64 system.

What do you mean? Have you installed WSLG on Win 10? How? Please, explain...

**Comment by Biswa96** [comment]
I will explain everything once it's done. In the screenshot, the Weston is _barely_ running in system.vhd file, not an user distribution. I also need some help in certain programming things. If anyone interested suggest a repository name and we can discuss it there.

**Comment by Dakkaron** [comment]
@Biswa96 Call it wslfg (f for fake)

**Comment by jimpriest** [comment]
This is a real drag for people using a Win10 machine in a work environment.  I'm currently running Win 10 Business and I imagine it will be quite a long time before my corporate IT update to Windows 11.

**Comment by Biswa96** [comment]
@jimpriest What programs would you like to run in Windows 10 if wslg works? I want to test in my Windows 10 test setup.

**Comment by crramirez** [comment]
Konsole, Terminator

**Comment by Biswa96** [comment]
Does those work in Windows 11? In my Windows 10 test setup, those show a window but unresponsive. Need to dive into logs.

**Comment by crramirez** [comment]
Yes, they work both of them. Try first with Terminator and remember it needs dbus-x11.

**Comment by multimokia** [comment]
What a disappointment. This was the only thing I was looking forward to in 21H2, given I don't intend to move to 11. Seeing the screenshots on a Windows 10 machine and the fact it was on a 10 insider channel implied it was coming to 10. What a waste.

**Comment by leebehrens** [comment]
Well this is disappointing. I was counting the days for 21H2 to come to my work laptop, so I could run all of VSCode in WSL2. 21H2 is on my system at last, and upon searching for how to enable WSLg I find my patience has been for naught. 😒

**Comment by Masamune3210** [comment]
This issue really has just became people complaining about it not being on 10, huh, There are architectural reasons that its not as easy as just plopping wslg into 10, there are more changes needed than just adding the runtime and tweaking the UI

**Comment by Dakkaron** [comment]
Over here (https://github.com/microsoft/wslg/issues/347#issuecomment-971126353) a Microsoft dev said, it wasn't too hard to implement it for Win10, and it might be implemented if there are enough upvotes/complaints. Now it's 5 months and a ton of upvotes later, and nothing.

I just got the OK from our helpdesk to switch my company PC over to Linux. Looks like I will be doing just that.

**Comment by Masamune3210** [comment]
You missed the crucial point in that comment that things have to be changed inside 10 itself first, which takes time

**Comment by jimpriest** [comment]
@Dakkaron  Yep. I really tried to make Windows work but it's just too painful. Eventually installed PopOS on my work laptop and haven't looked back.  Docker runs soooo fast.

**Comment by Dakkaron** [comment]
@Masamune3210 Deciding (and announcing) whether you are going to do it or not, doesn't. Noone is asking for it to be implemented right now. But it would be important to know if there is a purpose in waiting for that feature. 5 months is long enough to decide whether the public interest in this feature is worth the implementation effort.

**Comment by leebehrens** [comment]
I totally understand that Microsoft discovered the need for significant changes for WSLg to work correctly in Win10. I haven't really kept up on the topic over the last upwards of six months, because my work machine didn't have what I understood to be the necessary Win10 version until just recently.

My family's systems, though perfectly good and well-performing, will most likely be deemed officially unsupported by Windows 11. I need Linux, so Debian became the primary OS on my system, with Windows as secondary. The swap was surprisingly easy, though about as time-consuming as any fresh system wipe to get what I needed reinstalled. Though it's there should I need it, I can't remember the last time I fired up Windows on my own system.

**Comment by kevinmcfarlane** [comment]
This is one of Microsoft's worst habits. 

1. They introduce a new product/feature.
2. The preview or beta version is available in hosting environment version X. This could be Windows or Visual Studio, say.
3. The RTM version is released and then they make it available only for the new version of Windows or Visual Studio and not version X.
4. This is almost fraudulent since it is usually the case that there was at least a beta version of the new Windows or Visual Studio available, so they should have hosted the new feature/product in that to start off with. 

Re: WSLg I was quite keen to try this. I don't meet the requirements for upgrading to Windows 11. Many people will be in this situation.

I have been using WSL2 to attempt to learn blockchain development, most of which requires Linux or Mac to work properly. Unfortunately, you frequently come unstuck just using WSL2 (often mentioned by the crypto devs).
I've had several attempts at using VirtualBox and that has its own issues with startup freezes, mainly linked to guest additions I think (I'll probably have another go and just live without them, but it's been pretty demoralising).

Maybe Microsoft will reconsider?

**Comment by angelog0** [comment]
> My machine, based on Ryzen 5 and 8 GiB RAM, is only 3 years old and for what I read I do not have enough hardware to upgrade to Win 11. In any case, I never will upgrade to an OS which, for what I read, will request an MS account to complete its installation. I consider this limiting the user freedom. After all, this is a dual boot machine and it is not so hard to convert the Windows partition to a Linux partition so the the it becomes a full Linux machine...

**Alea iacta est!**

Windows 10 Home on one PC has gone, supplanted by a full GNU/Linux distribution.

Now there is another machine with Win 10 Pro...

**Comment by Masamune3210** [comment]
> This is one of Microsoft's worst habits. 
> 
> 1. They introduce a new product/feature.
> 2. The preview or beta version is available in hosting environment version X. This could be Windows or Visual Studio, say.
> 3. The RTM version is released and then they make it available only for the new version of Windows or Visual Studio and not version X.
> 4. This is almost fraudulent since it is usually the case that there was at least a beta version of the new Windows or Visual Studio available, so they should have hosted the new feature/product in that to start off with. 
> 
> Re: WSLg I was quite keen to try this. I don't meet the requirements for upgrading to Windows 11. Many people will be in this situation.
> 
> I have been using WSL2 to attempt to learn blockchain development, most of which requires Linux or Mac to work properly. Unfortunately, you frequently come unstuck just using WSL2 (often mentioned by the crypto devs).
> I've had several attempts at using VirtualBox and that has its own issues with startup freezes, mainly linked to guest additions I think (I'll probably have another go and just live without them, but it's been pretty demoralising).
> 
> Maybe Microsoft will reconsider? 

The thing that people keep missing about this is that they have said multiple times now that by the time that wslg was in the beta, it was basically only 10 in name, most of the things that would become 11 were already in place and being tested

**Comment by Dakkaron** [comment]
> > This is one of Microsoft's worst habits.
> > 
> > 1. They introduce a new product/feature.
> > 2. The preview or beta version is available in hosting environment version X. This could be Windows or Visual Studio, say.
> > 3. The RTM version is released and then they make it available only for the new version of Windows or Visual Studio and not version X.
> > 4. This is almost fraudulent since it is usually the case that there was at least a beta version of the new Windows or Visual Studio available, so they should have hosted the new feature/product in that to start off with.
> > 
> > Re: WSLg I was quite keen to try this. I don't meet the requirements for upgrading to Windows 11. Many people will be in this situation.
> > I have been using WSL2 to attempt to learn blockchain development, most of which requires Linux or Mac to work properly. Unfortunately, you frequently come unstuck just using WSL2 (often mentioned by the crypto devs).
> > I've had several attempts at using VirtualBox and that has its own issues with startup freezes, mainly linked to guest additions I think (I'll probably have another go and just live without them, but it's been pretty demoralising).
> > Maybe Microsoft will reconsider?
> 
> The thing that people keep missing about this is that they have said multiple times now that by the time that wslg was in the beta, it was basically only 10 in name, most of the things that would become 11 were already in place and being tested

And the thing you are missing is that a Microsoft guy said, it would be easy to backport to Win10 and they would do it if there is enough interest in it.

Apparently there wasn't, so they won't, so everyone is moving to Linux.

**Comment by queses** [comment]
IMO the WSLg feature is just not good enough to discuss it that much 🙂. I upgraded to Win11 and tried to use WSLg to run a Jetbrains IDE. WSLg is not perfomant enough, and the window decorations looks afwul. Feels sluggish and inconsistent. So I switched back to Linux anyway.

**Comment by hydino2085143** [comment]
windows 11 is literally the same thing as windows 10, except with a new dwm theme and neutered UI. There is no reason that wslg and it's dependencies cannot be backported to windows 10, except for the fact that ms wants to artificially increase adoption of a new theme engine for their os.

**Comment by Masamune3210** [comment]
Its really not, and besides that point that seemingly always gets lost, the theming update takes literally no extra effort, to the point that old apps get the new theming without even requiring a update if they have been made with anything even vaguely resembling a modern UI suite

**Comment by bersbersbers** [comment]
Looks like this is fixed, see https://github.com/microsoft/wslg/issues/567#issuecomment-1319599234. I can run `xeyes` on Windows 10 21H2 now (19044.2311, to be exact).
